whistleblower/targets/twitter.py

```diff
@@ -80,23 +80,38 @@ def provision_database(self):
         Persist database records for each of the already existing posts
         in the Twitter timeline.
         """
-        posts = map(self.__database_record_for_post, self.posts())
-        self.database.posts.insert_many(list(posts))
+        for posts_chunk in self.posts():
+            posts = map(self.__database_record_for_post, posts_chunk)
+            posts = [post for post in posts if post]
+            self.database.posts.insert_many(posts)
 
     def posts(self):
         """
-        Posts already in the Twitter timeline.
-        """
-        return self.api.GetUserTimeline(screen_name=self.PROFILE)
+        Posts already in the Twitter timeline. Paginates the results in lists
+        of 20 posts (given also to a Twitter API limitation).
+        """
+        max_id = None
+        while True:
+            posts = self.api.GetUserTimeline(screen_name=self.PROFILE,
+                                             max_id=max_id)
+            if max_id != None:
+                posts = posts[1:]
```

Comment:
Twitter API only allows reading 20 statuses per time. And when you're about to read the next 20 statuses, you get the `id` of the last status on the previous list and the next 19 statuses. So you need to discard the first one when reading a new list.